#### Задание 1.

Даны значения зарплат из выборки выпускников: 100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150. Посчитать (желательно без использования статистических методов наподобие std, var, mean) среднее арифметическое, среднее квадратичное отклонение, смещенную и несмещенную оценки дисперсий для данной выборки.

In [1]:
import numpy as np

a = np.array([100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 24, 57, 55, 70, 75, 65, 84, 90, 150])
a

array([100,  80,  75,  77,  89,  33,  45,  25,  65,  17,  30,  24,  57,
        55,  70,  75,  65,  84,  90, 150])

In [2]:
# Среднее арифметическое
my_mean = a.sum()/len(a)
print(my_mean, a.mean())

65.3 65.3


In [3]:
# Среднее квадратическое отклонение
my_std = 0
for itm in a:
    my_std += pow(itm - my_mean, 2)
my_std = np.sqrt(my_std / len(a))

print(my_std, a.std())

30.823854398825596 30.823854398825596


In [4]:
# Смещенная оценка дисперсии
my_var = 0
for itm in a:
    my_var += pow(itm - my_mean, 2)
my_var = my_var / len(a)

print(my_var, a.var())

950.11 950.11


In [5]:
# Несмещенная оценка дисперсии
my_var_1 = 0
for itm in a:
    my_var_1 += pow(itm - my_mean, 2)
my_var_1 = my_var_1 / (len(a) - 1)

print(my_var_1, a.var(ddof=1))

1000.1157894736842 1000.1157894736842


#### Задание 2.

В первом ящике находится 8 мячей, из которых 5 - белые. Во втором ящике - 12 мячей, из которых 5 белых. Из первого ящика вытаскивают случайным образом два мяча, из второго - 4. Какова вероятность того, что 3 мяча белые?

Благоприятными исходами при вытаскивании мячей из двух корзин будут следующие комбинации мячей:

- 1-я корзина - 2 белых, 2-я корзина - 1 белый, 3 черных

- 1-я корзина - 1 белый, 1 черный, 2-я корзина - 2 белых, 2 черных

- 1-я корзина - 2 черных, 2-я корзина - 3 белых, 1 черный

In [6]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [7]:
p1 = (combinations(5, 2)/combinations(8, 2))*(combinations(5, 1)*combinations(7, 3)/combinations(12, 4))
p2 = (combinations(5, 1)*combinations(3, 1)/combinations(8, 2))*(combinations(5, 2)*combinations(7, 2)/combinations(12, 4))
p3 = (combinations(3, 2)/combinations(8, 2))*(combinations(5, 3)*combinations(7, 1)/combinations(12, 4))
p = p1 + p2 + p3
print(p1, p2, p3, p)

0.12626262626262627 0.22727272727272727 0.01515151515151515 0.3686868686868687


Ответ: вероятность того, что 3 мяча белые - 36.87%

#### Задание 3.

На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. Найти вероятность того, что выстрел произведен: a). первым спортсменом б). вторым спортсменом в). третьим спортсменом.

Искомые вероятности найдем по формуле:

$$P(B_{n}|A)=\frac{P(B_{n})\cdot P(A|B_{n})}{P(A)}$$

$P(A|B_{n})$ - заданные вероятности попадания каждого спортсмена в мишень,

$P(B_{n})$ - вероятность выстрела каждого спортсмена, $P(B_{n})=\frac{1}{3}$,

$P(A)$ - вероятность попадания в мишень, $P(A)=\frac{1}{3}(0.9+0.8+0.6)=0.7(6)$

Таким образом,

$$P(B_{1}|A)=\frac{\frac{1}{3}\cdot 0.9}{0.7(6)}=39.13\%$$



$$P(B_{2}|A)=\frac{\frac{1}{3}\cdot 0.8}{0.7(6)}=34.78\%$$



$$P(B_{3}|A)=\frac{\frac{1}{3}\cdot 0.6}{0.7(6)}=26.09\%$$

#### Задание 4.

В университет на факультеты A и B поступило равное количество студентов, а на факультет C студентов поступило столько же, сколько на A и B вместе. Вероятность того, что студент факультета A сдаст первую сессию, равна 0.8. Для студента факультета B эта вероятность равна 0.7, а для студента факультета C - 0.9. Студент сдал первую сессию. Какова вероятность, что он учится: a). на факультете A б). на факультете B в). на факультете C?

Искомые вероятности найдем по формуле:

$$P(B_{n}|A)=\frac{P(B_{n})\cdot P(A|B_{n})}{P(A)}$$

$P(A|B_{n})$ - заданные вероятности сдать сессии в каждом факультете,

$P(B_{n})$ - вероятность, что случайно выбранный студент учится в данном факультете, $P(B_{1})=P(B_{2})=0.25,P(B_{3})=0.5$,

$P(A)$ - вероятность сдать сессию для случайного студента, $P(A)=0.25(0.8+0.7)+0.5\cdot 0.9=0.825$

Таким образом,


$$P(B_{1}|A)=\frac{0.25\cdot 0.8}{0.825}=24.24\%$$



$$P(B_{2}|A)=\frac{0.25\cdot 0.7}{0.825}=21.21\%$$



$$P(B_{3}|A)=\frac{0.5\cdot 0.9}{0.825}=54.54\%$$

#### Задание 5.

Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя: а). все детали б). только две детали в). хотя бы одна деталь г). от одной до двух деталей?

In [8]:
p1 = 0.1
p2 = 0.2
p3 = 0.25

# a)
p = p1 * p2 * p3
p

0.005000000000000001

In [9]:
# b)

p = p1*p2*(1-p3) + p1*p3*(1-p2) + p2*p3*(1-p1)
p

0.08000000000000002

In [10]:
# c)

q = (1-p1)*(1-p2)*(1-p3)
p = 1-q
p

0.45999999999999996

In [11]:
# d) от одной до 2-х деталей можно представить как вероятность выхода из строя хотя бы одной детали минус вероятность выхода
# из строя всех деталей

p = 1-q - p1 * p2 * p3
p

0.45499999999999996